BossMod/QuestBattle/QuestBattleDirector.cs

```diff
@@ -0,0 +1,375 @@
+﻿using BossMod.AI;
+using BossMod.Autorotation;
+using Dalamud.Plugin.Ipc;
+using System.Threading.Tasks;
+
+namespace BossMod.QuestBattle;
+
+class PathfindNoop : ICallGateSubscriber<Vector3, Vector3, bool, Task<List<Vector3>>?>
+{
+    public void InvokeAction(Vector3 arg1, Vector3 arg2, bool arg3) { }
+    public Task<List<Vector3>>? InvokeFunc(Vector3 arg1, Vector3 arg2, bool arg3) => null;
+    public void Subscribe(Action<Vector3, Vector3, bool> action) { }
+    public void Unsubscribe(Action<Vector3, Vector3, bool> action) { }
+}
+
+class PathReadyNoop : ICallGateSubscriber<bool>
+{
+    public void InvokeAction() { }
+    public bool InvokeFunc() => false;
+    public void Subscribe(Action action) { }
+    public void Unsubscribe(Action action) { }
+}
+
+public sealed class QuestBattleDirector : IDisposable
```

Comment:
Ok, now that we have ZoneModuleManager, I think everything here can be moved into an intermediate base class for quest modules:
```
public abstract class QuestZoneModule : ZoneModule { common stuff }
public class AFooInMyBar : QuestZoneModule { implementation }
```

Reply:
Btw feel free to subscribe to everything needed in base class (ZoneModule) and call virtual functions, I think it's easier to deal with by users than having to subscribe to shit manually...

Reply:
by "users" you mean devs?

Reply:
Yes